counterpartylib/lib/transaction.py

```diff
@@ -487,4 +491,199 @@ def construct (db, tx_info, encoding='auto',
+        # TODO: check all messages
+        # check destination
+        if asked_message_type == 'bet':
+            params['destination'] = params['feed_address']
+
+        # No destination
+        if asked_message_type in ['order', 'dividend', 'broadcast', 'cancel', 'destroy', 'execute', 'publish', 'rps', 'rpsresolve']:
+            if len(tx_info['destinations'].keys()) != 0:
+                raise exceptions.TransactionError('invalid destination')
+        # No destination or one destination
+        elif asked_message_type in ['issuance']:
+            if len(tx_info['destinations'].keys()) > 1:
+                raise exceptions.TransactionError('invalid destination')
+            if len(tx_info['destinations'].keys()) == 1 and params['transfer_destination'] not in tx_info['destinations']:
+                raise exceptions.TransactionError('invalid destination')
+        # only one destination
+        elif asked_message_type in ['send', 'bet']:
+            if len(tx_info['destinations'].keys()) != 1:
+                raise exceptions.TransactionError('invalid destination')
+            if params['destination'] not in tx_info['destinations']:
+                raise exceptions.TransactionError('invalid destination')
+            if tx_info['destinations'][params['destination']] > config.DEFAULT_MULTISIG_DUST_SIZE:
+                raise exceptions.TransactionError('invalid quantity for the destination')
+        # only one unknown destination and amount
+        elif asked_message_type == 'btcpay':
+            if len(tx_info['destinations'].keys()) != 1:
+                raise exceptions.TransactionError('invalid destination')
+
+        # check data value
+        max_data_value = 0
+        if tx_info['data_output_type'] == 'OP_CHECKSIG':
+            dust_size = params['regular_dust_size'] if 'regular_dust_size' in params and params['regular_dust_size'] else config.DEFAULT_REGULAR_DUST_SIZE
+            max_data_value = tx_info['data_output_count'] * dust_size
+        elif tx_info['data_output_type'] == 'OP_CHECKMULTISIG':
+            dust_size = params['multisig_dust_size'] if 'multisig_dust_size' in params and params['multisig_dust_size'] else config.DEFAULT_MULTISIG_DUST_SIZE
+            max_data_value = tx_info['data_output_count'] * dust_size
+        if tx_info['data_value'] > max_data_value:
+            raise exceptions.TransactionError('invalid dust quantity for data outputs')
+
+    else:
+        # only BTC sends don't contain data
+        if (method != 'create_burn') and (method != 'create_send' or params['asset'] != 'BTC'):
+            raise exceptions.TransactionError('constructed transaction does not contains data') 
+        if method == 'create_burn':
+            params['destination'] = config.UNSPENDABLE
+
+        # check change address (change in destinations)
+        if tx_info['change'][0] is not None:
+            raise exceptions.TransactionError('invalid change address')
+
+        # check destination
```

Comment:
All of these comments need to be more descriptive. Check destination *how*, *why*, *against what*?